tests/Http/HttpRequestTest.php

```diff
@@ -537,4 +537,22 @@ public function testCreateFromBase()
 
         $this->assertEquals($request->request->all(), $body);
     }
+
+    public function testJsonContent()
+    {
+        $body = [
+            'foo' => 'bar',
+            'baz' => ['qux'],
+        ];
+
+        $server = [
+            'CONTENT_TYPE' => 'application/json',
```

Comment:
There's a lot of "odd" in symfony, which I'd rather not get too far into (other than making this work for Laravel). Also the 'yoda conditions' littered through their code really bug me.